state/pruner.go

```diff
@@ -23,11 +24,14 @@ var (
 type Pruner struct {
 	service.BaseService
 	logger log.Logger
+
+	mtx sync.Mutex
```

Comment:
This was added after speaking to @mzabaluev today where we clearly want to serialize all requests that set any retain heights. For example, in the case where users may ignore our recommendations to only attach a single data companion and two calls are made simultaneously to set a particular retain height, allowing both calls to execute simultaneously may have undefined consequences.

Reply:
We discussed this initially and were hoping to rely on the databases' guarantees that these conflicts should be handled. But I agree this is a much cleaner approach, especially given the fact that we support multiple DB backends atm.